projects/cashmere/src/lib/datepicker/datepicker-input/datepicker-input.directive.ts

```diff
@@ -323,6 +323,22 @@ export class DatepickerInputDirective implements ControlValueAccessor, OnDestroy
 
     _onInput(value: string) {
         let date = this._dateAdapter.parse(value, this._dateFormats.parse.dateInput);
+        /** Two-digit year input conversion method for IE
+         * Based on the current year, assume that the four-digit year date should be in
+         * either the next 30 years, or the preceding 70 years */
+        if (date) {
+            let inputString: string = this._elementRef.nativeElement.value;
+            /** Skip this check if the input string contains any 3+ digit numerical values - assumed to be a year */
+            if (!inputString.match('[1-9][0-9][0-9]')) {
```

Comment:
This works, but it might be clearer to write it as a regular expression. `/[1-9][0-9][0-9]/`

Reply:
Gah - good catch.  I screwed up my syntax there, meant it to be a regular expression but wrote it wrong.  Just pushed a commit to correct that.  Thanks @isaaclyman